lib/byron/test/integration/Main.hs

```diff
@@ -0,0 +1,157 @@
+    , unsafeRequest
+    )
+
+import qualified Cardano.Wallet.Api.Link as Link
+import qualified Data.Aeson as Aeson
+import qualified Data.Text as T
+import qualified Test.Integration.Scenario.API.ByronTransactions as ByronTransactions
+import qualified Test.Integration.Scenario.API.ByronWallets as ByronWallets
+import qualified Test.Integration.Scenario.API.Network as Network
+
+-- | Define the actual executable name for the bridge CLI
+instance KnownCommand Byron where
+    commandName = "cardano-wallet-byron"
+
+main :: forall t. (t ~ Byron) => IO ()
+main = withUtf8Encoding $ withLogging Nothing Info $ \(_, tr) -> do
+    hspec $ do
+        describe "API Specifications" $ specWithServer tr $ do
+            ByronWallets.spec
+            ByronTransactions.spec
+            Network.spec
+
+specWithServer
+    :: Trace IO Text
+    -> SpecWith (Context Byron)
+    -> Spec
+specWithServer tr = aroundAll withContext . after tearDown
+  where
+    withContext :: (Context Byron -> IO ()) -> IO ()
+    withContext action = do
+        ctx <- newEmptyMVar
+        let setupContext bp wAddr = do
+                let baseUrl = "http://" <> T.pack (show wAddr) <> "/"
+                logInfo tr baseUrl
+                let sixtySeconds = 60*1000*1000 -- 60s in microseconds
+                manager <- (baseUrl,) <$> newManager (defaultManagerSettings
+                    { managerResponseTimeout =
+                        responseTimeoutMicro sixtySeconds
+                    })
+                faucet <- initFaucet
+                putMVar ctx $ Context
+                    { _cleanup = pure ()
+                    , _manager = manager
+                    , _walletPort = Port . fromIntegral $ unsafePortNumber wAddr
+                    , _faucet = faucet
+                    , _feeEstimator = mkFeeEstimator
+                    , _blockchainParameters = bp
+                    , _target = Proxy
+                    }
+        race (takeMVar ctx >>= action) (withServer setupContext) >>=
+            either pure (throwIO . ProcessHasExited "integration")
+
+    withServer action =
+        withCardanoNode tr $ \addrInfo block0 (bp,vData) ->
+        withSystemTempDirectory "cardano-wallet-databases" $ \db -> do
+            serveWallet @'Mainnet
+                (setupTracers (tracerSeverities (Just Info)) tr)
+                (SyncTolerance 10)
+                (Just db)
+                "127.0.0.1"
+                ListenOnRandomPort
+                addrInfo
+                block0
+                (bp, vData)
+                (action bp)
+
+    -- | teardown after each test (currently only deleting all wallets)
+    tearDown :: Context t -> IO ()
+    tearDown ctx = do
+        (_, wallets) <- unsafeRequest @[ApiByronWallet] ctx
+            (Link.listWallets @'Byron) Empty
+        forM_ wallets $ \w -> void $ request @Aeson.Value ctx
+            (Link.deleteWallet @'Byron w) Default Empty
+
+-- FIXME: Write fee estimator for Byron nodes.
```

Comment:
so here, we cannot just reuse https://github.com/input-output-hk/cardano-wallet/blob/master/lib/jormungandr/test/integration/Main.hs#L219 because it is going to be different, right?

Reply:
Exactly! To save up some time, I've just provided a dummy implementation for now and will provide a real one (resurrected fromthe http-bridge) in #1350